src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs

```diff
@@ -943,7 +890,7 @@ private async Task<bool> OnReceiveDebuggerAgentEvent(SessionId sessionId, JObjec
             return false;
         }
 
-        private async Task<MethodInfo> LoadSymbolsOnDemand(AssemblyInfo asm, int method_token, SessionId sessionId, CancellationToken token)
+        internal async Task<MethodInfo> LoadSymbolsOnDemand(AssemblyInfo asm, int method_token, SessionId sessionId, CancellationToken token)
         {
             ExecutionContext context = GetContext(sessionId);
             if (urlSymbolServerList.Count == 0)
```

Comment:
I don't think so, this is done on MonoSdbHelper line 708, pdb information will be already updated if it was found on LoadSymbolsOnDemand, so it will not enter on LoadSymbolsOnDemand. In the case that it will enter again, it's because couldn't find debug information, so it should really return null.